client/allochealth/tracker.go

```diff
@@ -68,7 +68,7 @@ type Tracker struct {
 
 	// lifecycleTasks is a set of tasks with lifecycle hook set and may
 	// terminate without affecting alloc health
-	lifecycleTasks map[string]bool
+	lifecycleTasks map[string]string
```

Comment:
Need to update the comment as it's now a mapping, not a set. Something like:

```
// lifecycleTasks are a map of lifecycle tasks to their hook type.
```

Is the `Lifecycle tasks may terminate without affect alloc health` part true? Seems like sidecar tasks should affect health?

Reply:
when the map is initialized, it does not include sidecar tasks. will change the comment to better reflect its usage.